catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -13,200 +13,798 @@
+
+        List<Filter> idFilters = new ArrayList<>();
+        idFilters.addAll(ids.stream()
+                .map(id -> filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .is()
+                        .equalTo()
+                        .text(id))
+                .collect(Collectors.toList()));
+        List<Filter> filters = getBasicFilter();
+        filters.add(filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE)
+                .is()
+                .bool(true));
+        Filter filter = filterBuilder.allOf(filters);
+        filter = filterBuilder.allOf(filter, filterBuilder.anyOf(idFilters));
+        return getRegistryMetacardsByFilter(filter);
     }
 
     @Override
-    public void updateRegistryEntry(String xml, Set<String> destinations)
+    public List<Metacard> getRegistryMetacardsByRegistryIds(List<String> ids)
             throws FederationAdminException {
-
+        if (CollectionUtils.isEmpty(ids)) {
+            throw new FederationAdminException(
+                    "Error getting registry metacards by registry ids. Null list of Ids provided.");
+        }
+
+        List<Filter> idFilters = new ArrayList<>();
+        idFilters.addAll(ids.stream()
+                .map(id -> filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .is()
+                        .equalTo()
+                        .text(id))
+                .collect(Collectors.toList()));
+
+        List<Filter> filters = getBasicFilter();
+        Filter filter = filterBuilder.allOf(filters);
+        filter = filterBuilder.allOf(filter, filterBuilder.anyOf(idFilters));
+        return getRegistryMetacardsByFilter(filter);
     }
 
     @Override
-    public List<Metacard> getRegistryMetacards() throws FederationAdminException {
-        return new ArrayList<>();
+    public List<RegistryPackageType> getLocalRegistryObjects() throws FederationAdminException {
+        List<RegistryPackageType> registryEntries = new ArrayList<>();
+
+        for (Metacard metacard : getLocalRegistryMetacards()) {
+            String xml = metacard.getMetadata();
+            registryEntries.add(getRegistryPackageFromString(xml));
+        }
+        return registryEntries;
     }
 
     @Override
-    public List<Metacard> getLocalRegistryMetacards() throws FederationAdminException {
-        return new ArrayList<>();
+    public List<RegistryPackageType> getRegistryObjects() throws FederationAdminException {
+        List<RegistryPackageType> registryEntries = new ArrayList<>();
+        for (Metacard metacard : getRegistryMetacards()) {
+            String xml = metacard.getMetadata();
+            registryEntries.add(getRegistryPackageFromString(xml));
+        }
+        return registryEntries;
     }
 
     @Override
-    public List<Metacard> getRegistryMetacardsByRegistryIds(List<String> ids)
+    public RegistryPackageType getRegistryObjectByRegistryId(String registryId)
```

Comment:
Actually, as a general rule, we try to avoid `null` altogether. Can we avoid it here instead of sometimes returning `null` and sometimes throwing exceptions?